src/openvic-simulation/map/Map.cpp

```diff
@@ -644,3 +646,89 @@ bool Map::generate_and_load_province_adjacencies(std::vector<LineObject> const&
 	);
 	return ret;
 }
+
+bool Map::load_climate_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary_reserve_length(climates, [this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid climate identifier - empty!");
+			return false;
+		}
+
+		bool ret = true;
+		Climate* cur_climate = climates.get_item_by_identifier(identifier);
+		if (cur_climate == nullptr) {
+			ModifierValue values;
+			ret &= modifier_manager.expect_modifier_value(move_variable_callback(values))(node);
+			ret &= climates.add_item({ identifier, std::move(values) });
+		} else {
+			ret &= expect_list(expect_province_identifier([this, &cur_climate, &identifier](Province& province) {
+				if (province.climate != cur_climate) {
+					cur_climate->add_province(&province);
+					if (province.climate != nullptr) {
+						Climate* old_climate = const_cast<Climate*>(province.climate);
+						old_climate->remove_province(&province);
+						Logger::warning(
+							"Province with id ", province.get_identifier(),
+							" found in multiple climates: ", identifier,
+							" and ", old_climate->get_identifier()
+						);
+					}
+					province.climate = cur_climate;
+				} else {
+					Logger::warning(
+						"Province with id ", province.get_identifier(),
+						"defined twice in climate ", identifier
+					);
+				}
+				return true;
+			}))(node);
+			cur_climate->lock();
+		}
+		return ret;
+	})(root);
+
+	lock_climates();
+
+	return ret;
+}
+
+bool Map::load_continent_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary_reserve_length(continents, [this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid continent identifier - empty!");
+			return false;
+		}
+
+		ModifierValue values;
+		ProvinceSetModifier::provinces_t prov_list;
+		bool ret = modifier_manager.expect_modifier_value_and_keys(move_variable_callback(values),
+			"provinces", ONE_EXACTLY, expect_list(expect_province_identifier(
+				[this, &prov_list, &identifier](Province const& province) -> bool {
```

Comment:
```suggestion
			"provinces", ONE_EXACTLY, expect_list_reserve_length(prov_list, expect_province_identifier(
				[&prov_list](Province const& province) -> bool {
```
Again, `expect_list_reserve_length` can be used, this time reserving space in a `std::vector` but it still works the same. Neither `this` nor `&identifier` are needed in the lambda capture.